services/website-api/src/controller/authentication/AuthenticationController.ts

```diff
@@ -0,0 +1,59 @@
+import { AuthenticationService } from '@donategifts/authentication';
+import { Controller, Post, Route, Tags, Response, Body } from '@tsoa/runtime';
+import { IUser, ObjectId } from '@donategifts/common';
+// import * as express from 'express';
+import { IAPIUser } from '../user/types/IAPIUser';
+import { IAPISignupResponse } from './types/IAPISignupResponse';
+
+@Route('/auth')
+@Tags('authentication')
+export class AuthenticationController extends Controller {
+	constructor(private authenticationService: typeof AuthenticationService = AuthenticationService) {
+		super();
+	}
+
+	@Response('400', 'Bad request')
+	@Post('/signup')
+	public async signup(
+		@Body() body: { userData: IAPIUser; captchaToken: string },
+	): Promise<IAPISignupResponse> {
+		return this.authenticationService.signupUser(
+			{
+				...body.userData,
+				_id: ObjectId<IUser>(body.userData._id),
+			},
+			body.captchaToken,
+		);
+	}
+
+	@Response('400', 'Bad request')
+	@Post('/google')
+	public async google(@Body() body: { idToken: string }): Promise<IAPIUser> {
+		return this.authenticationService.googleSignIn(body.idToken);
+	}
+
+	@Response('400', 'Bad request')
+	@Post('/facebook')
+	public async facebook(@Body() body: { userName: string; email: string }): Promise<IAPIUser> {
+		return this.authenticationService.facebookSignIn(body.userName, body.userName);
+	}
+
+	@Response('400', 'Bad request')
+	@Post('/login')
+	public async login(
+		@Body() body: { email: string; password: string },
```

Comment:
why the inconsistent format? methods above have longer function params and are not broken up.

Reply:
ask eslint, maybe the line width? idk